source/common/quic/envoy_quic_server_session.cc

```diff
@@ -44,8 +44,10 @@ quic::QuicSpdyStream* EnvoyQuicServerSession::CreateIncomingStream(quic::QuicStr
     return nullptr;
   }
   if (!codec_stats_.has_value() || !http3_options_.has_value()) {
-    ENVOY_CONN_LOG(error, "Attempt to create stream {} before HCM filter is initialized.", *this,
-                   id);
+    ENVOY_BUG(false,
```

Comment:
Do you have a test for this ENVOY_BUG? If not just make it an assert.

Reply:
As long as our implementation is correct, it shouldn't be reached. So I don't know how to trigger it in a meaningful test right now. But I'd like to keep it in production, in case we miss some QUICHE behavior change that could occassionally lead to this in the future.

Reply:
But how will you know the error checking will work if we reach it in production? :) In general we require testing for ENVOY_BUG type logic, so I would still prefer that you switch this to assert if you can't reasonably test it.

Reply:
Okay, I added a unit test in envoy_quic_server_session_test to hit this ENVOY_BUG just to check the ENVOY_BUG works as intended in debug and release build.